src/transformStateWithClones.js

```diff
@@ -7,7 +7,47 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const totalPropertys = [];
+  let copyState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        copyState = {
+          ...copyState,
+          ...action.extraData,
+        };
+
+        totalPropertys.push({
+          ...copyState,
+          ...action.extraData,
+        });
```

Comment:
You can push your copyState to result once at the end, after switch loop, .push({ ...copyState })
And you first change copyState, then repeat these added items for push? You can use changed copyState for this

Reply:
yes, you are right